azuki-script-generation/src/main/kotlin/com/anaplan/engineering/azuki/script/generation/ScriptGenerationEnvironment.kt

```diff
@@ -0,0 +1,47 @@
+package com.anaplan.engineering.azuki.script.generation
+
+/**
+ * Environment persisted through the script generation process.
+ */
+interface ScriptGenerationEnvironment {
+
+    /**
+     * The list of checks produced from composing ComposableChecks through this environment.
+     * Having been composed through the environment, they do not need access to it again.
+     */
+    val composedChecks: List<BasicScriptGenerationCheck>
+}
+
+/**
+ * Dummy environment for users that don't need one.
+ */
+object NoScriptGenerationEnvironment : ScriptGenerationEnvironment {
+
+    override val composedChecks = listOf<BasicScriptGenerationCheck>()
```

Comment:
⛏️ Using `emptyList` is functionally the same, just a little bit more readable.
```suggestion
    override val composedChecks = emptyList<BasicScriptGenerationCheck>()
```

Reply:
I did wonder whether I should be using that.  I'll adopt it from now on - there have been a lot of commits I've made recently where I've used `listOf` and the analogous other collection constructors...